### PR TITLE
Improved template inheritance

### PR DESCRIPTION
**Pull request by KES777**
The main idea is to allow 'content' helper get arguments. 

This makes inheritance more flexible and powerful

### CODE REVIEW

**Comment by kraih** [comment]
Afraid i have to vote :-1: on this. The proposed changes add a lot of complexity to an already complex feature, and make the helpers `content`, `content_for` and `content_with` behave inconsistently. Also, i'm quite sure this suffers from the same exception handling problems as sharing normal reusable template blocks between multiple templates.

**Comment by kraih** [comment]
The exception handling problem i was referring to can be demonstrated with this app.
```
use Mojolicious::Lite;

get '/' => 'index';

app->start;
__DATA__

@@ index.html.ep
% layout 'default';
% stash->{myblock} = begin
   % die;
% end

@@ layouts/default.html.ep
%= stash->{myblock}()
```
You'll notice that there's only an error message and that the code for `index.html.ep` cannot be shown on the debug page.

File: lib/Mojolicious/Plugin/DefaultHelpers.pm [hunk 1]
```diff
@@ -49,20 +49,34 @@ sub _asset {
...
   if (defined $content) {
-    if ($append) { $hash->{$name} .= _block($content) }
-    if ($replace) { $hash->{$name} = _block($content) }
-    else          { $hash->{$name} //= _block($content) }
+    if ($append) { push @{$hash->{$name}}, $content }
+    if ($replace) { $hash->{$name} = [_block($content, @args)] }
+    else          { $hash->{$name} //= [$content] }
+  }
+
+  return '' if !defined $hash->{$name} || !defined wantarray;
```
Comment (KES777) [inline lib/Mojolicious/Plugin/DefaultHelpers.pm]: This line is just speed improvement. It can be safely removed in prejudice of speed.

Comment (KES777) [inline lib/Mojolicious/Plugin/DefaultHelpers.pm]: This line is just speed improvement. It can be safely removed in prejudice of speed.

**Comment by KES777** [comment]
I fix code. Now it look less complex.

**Comment by kraih** [comment]
Closing this issue since it didn't get the required votes.

**Pull request closed without merging**